src/course-unit/data/thunk.js

```diff
@@ -51,32 +52,53 @@ export function fetchCourseUnitQuery(courseId) {
   };
 }
 
-export function fetchCourseSectionVerticalData(courseId) {
+export function fetchCourseSectionVerticalData(courseId, sequenceId) {
   return async (dispatch) => {
     dispatch(updateLoadingCourseSectionVerticalDataStatus({ status: RequestStatus.IN_PROGRESS }));
+    dispatch(fetchSequenceRequest({ sequenceId }));
 
     try {
       const courseSectionVerticalData = await getCourseSectionVerticalData(courseId);
       dispatch(fetchCourseSectionVerticalDataSuccess(courseSectionVerticalData));
       dispatch(updateLoadingCourseSectionVerticalDataStatus({ status: RequestStatus.SUCCESSFUL }));
+      dispatch(updateModel({
+        modelType: 'sequences',
+        model: courseSectionVerticalData.sequence,
+      }));
+      dispatch(updateModels({
+        modelType: 'units',
+        models: courseSectionVerticalData.units,
+      }));
+      dispatch(fetchSequenceSuccess({ sequenceId }));
       return true;
     } catch (error) {
       dispatch(updateLoadingCourseSectionVerticalDataStatus({ status: RequestStatus.FAILED }));
+      dispatch(fetchSequenceFailure({ sequenceId }));
       return false;
     }
   };
 }
 
-export function editCourseItemQuery(itemId, displayName) {
+export function addNewSequenceNavigationUnit(unitId, sequenceId) {
   return async (dispatch) => {
     dispatch(updateSavingStatus({ status: RequestStatus.PENDING }));
-    dispatch(showProcessingNotification(NOTIFICATION_MESSAGES.saving));
+    dispatch(showProcessingNotification(NOTIFICATION_MESSAGES.adding));
 
     try {
-      await editUnitDisplayName(itemId, displayName).then(async (result) => {
+      await sendNewSequenceNavigationUnit(sequenceId).then(async (result) => {
```

Comment:
Either
```
const result = await sendNewSequenceNavigationUnit(sequenceId)
if (result) {
....
```
Or 
```
sendNewSequenceNavigationUnit(sequenceId).then((result) => {
   ...
}));
```
Both are fine but not together at the same time. I saw this in other places. I prefer to refactor such places when we copy or write own functions.

Reply:
At the moment this function has been removed, we are using the approach that is in the `ts-develop` branch